 <b>Import StopWords List, Positive and Negative words lists and join them as three strings </b>

In [486]:
with open('stopWords.txt') as stopWordsFile:
    stopWords ="".join(line.replace('\n',' ') for line in stopWordsFile)

In [487]:
with open('pWords.txt') as pWordsFile:
    pWords ="".join(line.replace('\n',' ') for line in pWordsFile)

In [488]:
with open('nWords.txt') as nWordsFile:
    nWords ="".join(line.replace('\n',' ') for line in nWordsFile)

In [489]:
stopWordsFile.close()
pWordsFile.close()
nWordsFile.close()

In [490]:
type(stopWords)
type(pWords)
type(nWords)


str

### Convert them to lists for later use

In [491]:
stopWordsList = list(stopWords.split(' '))

In [492]:
pList = list(pWords.split(' '))

In [493]:
nList = list(nWords.split(' '))

### Inspect first 5 elements

In [494]:
nList[0:5]

['2-faced', '2-faces', 'abnormal', 'abolish', 'abominable']

In [495]:
pList[0:5]

['a+', 'abound', 'abounds', 'abundance', 'abundant']

In [496]:
stopWordsList[0:5]

['a', 'able', 'about', 'across', 'after']

### Open scrapped txt file which has the word "Purdue" as the key word
#### lower the string for later comparison with words lists

In [497]:
with open('Purdue.txt') as inputFile:
    data="".join(line.strip().lower().decode('utf-8') for line in inputFile)

In [498]:
inputFile.close()

### Insepect data and we need to do the following
#### 1. Remove links
#### 2. Remove "rt and its following user name:"
#### 3. Remove "@username:"
#### 4. Remove all punctionation excepet apstrophe
##### Without above operations, the final score is about the same because these removes don't have much impact on the legitimate words we are comparing

In [499]:
data

u'rt @iuwsoccer: ht: #iuws leads purdue, 1-0. https://t.co/ievu4dy6mvrt @nickbaumgardner: john o\'korn felt he came full circle today. tony levine, the hc who benched him at houston, is purdue\'s co-oc.rt @umichfootball: total yards in the second half:michigan: 292 (21 points)purdue: 10 (0 points)#goblue https://t.co/cktwtvkslort @footballscoop: purdue is 0 for 11 3rd down conversions.don brown. gangsta. https://t.co/uv2v8iaomqrt @umichfootball: total yards in the second half:michigan: 292 (21 points)purdue: 10 (0 points)#goblue https://t.co/cktwtvkslort @nickbaumgardner: total yards in the third q:michigan 147, purdue -4rt @nickbaumgardner: second half yardsmichigan 274, purdue 1rt @drewchallett: michigan outscored purdue, 21-0, and outgained them, 292-10, in the second half. that was the most complete half u-m has\u2026@goblueinsider he didn\'t even perpare with the starters...or like a starter.....he could have put up 40+ on purdue if okorn started the gamd@kingklipsch @purdue they 

### Remove the links

In [500]:
import re

In [501]:
data = re.sub('https://t.co/[a-zA-Z0-9]{12}',' ', data)
data = re.sub('https://t.co/[a-zA-Z0-9]{10}',' ', data)

### Some links are still in the string
##### https:
##### https:/
##### https:/t
##### https:/t.
##### https:/t.c
##### https:/t.co
##### https://t
##### https://t.
##### https://t.co/21

In [502]:
data = re.sub('https:[/t.co/21]*',' ',data)


### Remove "rt @xxxxxx:" and "@xxxxx:"

In [503]:
data = re.sub('[rt]* @[a-zA-Z0-9_]+:',' ',data)

### Remove all punctuation except apostrophe

In [504]:
import string

regex = re.compile('[%s]' % re.escape(string.punctuation).replace("'",""))

In [505]:
data = regex.sub(' ', data)

### Remove all extra whitespaces

In [506]:
data = re.sub('[ ]+',' ', data)

In [507]:
data

u" ht iuws leads purdue 1 0 john o'korn felt he came full circle today tony levine the hc who benched him at houston is purdue's co oc total yards in the second half michigan 292 21 points purdue 10 0 points goblue purdue is 0 for 11 3rd down conversions don brown gangsta total yards in the second half michigan 292 21 points purdue 10 0 points goblue total yards in the third q michigan 147 purdue 4 second half yardsmichigan 274 purdue 1 michigan outscored purdue 21 0 and outgained them 292 10 in the second half that was the most complete half u m has\u2026 goblueinsider he didn't even perpare with the starters or like a starter he could have put up 40 on purdue if okorn started the gamd kingklipsch purdue they suck second half yardsmichigan 274 purdue 1 john o'korn felt he came full circle today tony levine the hc who benched him at houston is purdue's co oc purdue is 0 for 11 3rd down conversions don brown gangsta author sports news fresnobee com chris evans ran for two touchdowns in 

### Convert the string to a list

In [508]:
result = data.strip(' ').split(' ')

In [509]:
result[0:100]

[u'ht',
 u'iuws',
 u'leads',
 u'purdue',
 u'1',
 u'0',
 u'john',
 u"o'korn",
 u'felt',
 u'he',
 u'came',
 u'full',
 u'circle',
 u'today',
 u'tony',
 u'levine',
 u'the',
 u'hc',
 u'who',
 u'benched',
 u'him',
 u'at',
 u'houston',
 u'is',
 u"purdue's",
 u'co',
 u'oc',
 u'total',
 u'yards',
 u'in',
 u'the',
 u'second',
 u'half',
 u'michigan',
 u'292',
 u'21',
 u'points',
 u'purdue',
 u'10',
 u'0',
 u'points',
 u'goblue',
 u'purdue',
 u'is',
 u'0',
 u'for',
 u'11',
 u'3rd',
 u'down',
 u'conversions',
 u'don',
 u'brown',
 u'gangsta',
 u'total',
 u'yards',
 u'in',
 u'the',
 u'second',
 u'half',
 u'michigan',
 u'292',
 u'21',
 u'points',
 u'purdue',
 u'10',
 u'0',
 u'points',
 u'goblue',
 u'total',
 u'yards',
 u'in',
 u'the',
 u'third',
 u'q',
 u'michigan',
 u'147',
 u'purdue',
 u'4',
 u'second',
 u'half',
 u'yardsmichigan',
 u'274',
 u'purdue',
 u'1',
 u'michigan',
 u'outscored',
 u'purdue',
 u'21',
 u'0',
 u'and',
 u'outgained',
 u'them',
 u'292',
 u'10',
 u'in',
 u'the',
 u'second',
 u'hal

### Initial Score for Positive Words and Negative Words

In [510]:
negativeScore = 0
positiveScore = 0

### Go through a for loop and check if words are not in the StopWords List but in Positive and Negative lists

In [511]:
for word in dataList:
    if word not in stopWordsList:
        if word in pList:
            positiveScore += 1
        if word in nList:
            negativeScore -= 1
        

In [512]:
negativeScore

-71

In [513]:
positiveScore

176

In [514]:
totalSum = negativeScore + positiveScore

In [515]:
totalSum

105

### The total sum is 105 for this text. 
### The general sentiment is positive because the sum is positive.
### Since the positive score is twice more than the absolute value of the negative score, I'd  say the positive sentiment is <i>STRONG</i>
